src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -509,11 +509,7 @@ void SpatialPooler::compute(SDR &input, bool learn, SDR &active) {
   calculateOverlap_(input, overlaps_);
   calculateOverlapPct_(overlaps_, overlapsPct_);
 
-  if (learn) {
-    boostOverlaps_(overlaps_, boostedOverlaps_);
-  } else {
-    boostedOverlaps_.assign(overlaps_.begin(), overlaps_.end());
-  }
+  boostOverlaps_(overlaps_, boostedOverlaps_);
```

Comment:
this has been brought up before, should one do boosting even if learning is off? It does make sense, on the other hand, say you 
- train SP, (1)
- turn learning OFF, run some data
- you'd expect you return to the state (1) and can continue... This might be tested in #194 

It boils down to: can we run inference in a "const" state w/o changing the state of the algorithm? Maybe add a 3rd state (learn = LEARN/INFER/NO_CHANGES) ?